src/compas/scene/scene.py

```diff
@@ -108,19 +113,49 @@ def add(self, item, parent=None, **kwargs):
             The scene object associated with the item.
         """
 
-        parent = parent or self.root
+        if "context" in kwargs:
+            if kwargs["context"] != self.context:
+                raise Exception("Object context should be the same as scene context: {} != {}".format(kwargs["context"], self.context))
+            del kwargs["context"]  # otherwist the SceneObject receives "context" twice, which results in an error
+
+        # Create a corresponding new scene object
+        sceneobject = SceneObjectFactory.create(item=item, context=self.context, scene=self, **kwargs)
 
-        if isinstance(item, SceneObject):
-            sceneobject = item
+        # Add the scene object and item to the data store
+        self.objectstore[str(sceneobject.guid)] = sceneobject
+        self.datastore[str(item.guid)] = item
+
+        # Add the scene object to the hierarchical tree
+        if parent is None:
+            parent_node = self.tree.root
         else:
-            if "context" in kwargs:
-                if kwargs["context"] != self.context:
-                    raise Exception("Object context should be the same as scene context: {} != {}".format(kwargs["context"], self.context))
-                del kwargs["context"]  # otherwist the SceneObject receives "context" twice, which results in an error
-            sceneobject = SceneObject(item=item, context=self.context, **kwargs)  # type: ignore
-        super(Scene, self).add(sceneobject, parent=parent)
+            if not isinstance(parent, SceneObject):
+                raise ValueError("Parent is not a SceneObject.", parent)
+            parent_node = self.tree.get_node_by_name(parent.guid)
```

Comment:
should we not add a method for finding a node "by guid" instead of misusing the `get_node_by_name` method?

getting something by name to me implies that there might be multiple objects that might be found/returned.

Reply:
perhaps `tree.get(guid)` exists as default, whereas `tree.get_by_...()` provides alternatives

Reply:
I understand, using `name` can sound strange indeed. But if we use something like `get_by_guid`, it is also not entirely right, because this `guid` here we use for querying is the `guid` of a `SceneObject`, not the `guid` of the `TreeNode` itself, they are still two different objects of distinct classes, It's also weird for them to share the same `guid`.

Reply:
The most appropriate naming in my opinion would be `sceneobject_guid`, but to achieve that we either: 
1. store them in the built-in `attributes` dict of `Data` class. The downside would be adding an extra "attribute" dict layer in every node in serialization. 
2. Add a custom `SceneNode` class like what we did before, we can still make the serialization compact, with the price of adding another class that does only this.
3. Other better ideas? ...

Reply:
add a wrapper method to the scene itself, instead of using the method of the tree directly?

Reply:
ok, sounds like a pretty good mid point! will do this way!